planner/core/testdata/plan_suite_unexported_in.json

```diff
@@ -378,7 +378,10 @@
       "select count(1) from (select count(1), a as b from t group by a) tt group by b",
       "select a, count(b) from t group by a",
       "select a, count(distinct a, b) from t group by a",
-      "select a, approx_count_distinct(a, b) from t group by a"
+      "select a, approx_count_distinct(a, b) from t group by a",
+      // Test issue23436
+      "select count(distinct a) from t",
+      "select count(distinct a), sum(distinct a) from t"
```

Comment:
We can add cases like
- count(distinct a), sum(distinct b) where a, b are both unique & not null
- count(distinct x), where x is unique but can be null
- count(distinct a), avg(b) --> a is unique and not null while sum(b) is a normal aggregate function without distinct constraint.